### PR TITLE
Remove get with templates from state / components.

### PR DESCRIPTION
**Pull request by sgomes**
Using template strings with `_.get` is unsafe, since the values being interpolated could include characters that end up changing the path structure, such as `.` or `[]`.

#### Changes proposed in this Pull Request

* Remove _.get with string templates from components and state.

#### Testing instructions

No specific testing instructions, as this covers a wide range of functionality.

### CODE REVIEW

**Review by tyxla — APPROVED** [review]
LGTM 🚢 

It feels like we're too safe in some places (I've offered some suggestions) but as we start modularizing state, it may turn out to be a good idea.

File: client/state/comments/selectors.js [hunk 1]
```diff
@@ -27,13 +27,14 @@ import { getStateKey, deconstructStateKey, getErrorKey } from './utils';
...
  * Gets comment items for post
+ *
  * @param {object} state redux state
  * @param {number} siteId site identification
  * @param {number} postId site identification
  * @returns {Array} comment items
  */
 export const getPostCommentItems = ( state, siteId, postId ) =>
-	get( state.comments.items, `${ siteId }-${ postId }` );
+	state.comments.items?.[ `${ siteId }-${ postId }` ];
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/comments/selectors.js]: Fixed.

File: client/state/comments/selectors.js [hunk 2]
```diff
@@ -62,13 +63,14 @@ export const getCommentErrors = state => {
...
  * Get total number of comments on the server for a given post
+ *
  * @param {object} state redux state
  * @param {number} siteId site identification
  * @param {number} postId site identification
  * @returns {number} total comments count on the server. if not found, assume infinity
  */
 export const getPostTotalCommentsCount = ( state, siteId, postId ) =>
-	get( state.comments.totalCommentsCount, `${ siteId }-${ postId }` );
+	state.comments.totalCommentsCount?.[ `${ siteId }-${ postId }` ];
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/comments/selectors.js]: Fixed.

[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/comments/selectors.js]: Fixed.

File: client/state/domains/management/reducer.js [hunk 3]
```diff
@@ -158,9 +158,12 @@ export const items = withSchemaValidation( domainWhoisSchema, ( state = {}, acti
 		}
 		case DOMAIN_MANAGEMENT_WHOIS_UPDATE: {
 			const { domain, whoisData } = action;
-			const domainState = get( state, [ `${ domain }` ], {} );
+			const domainState = state?.[ domain ];
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/domains/management/reducer.js]: Fixed.

File: client/state/login/selectors.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ export const getTwoFactorUserId = state => {
  * @returns {?string}         The nonce.
  */
 export const getTwoFactorAuthNonce = ( state, nonceType ) => {
-	return get( state, `login.twoFactorAuth.two_step_nonce_${ nonceType }`, null );
+	return state?.login?.twoFactorAuth?.[ `two_step_nonce_${ nonceType }` ] ?? null;
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/login/selectors.js]: Fixed.

File: client/state/my-sites/sidebar/selectors.js [hunk 5]
```diff
@@ -1,8 +1,2 @@
-/**
- * External dependencies
- */
-
-import { get } from 'lodash';
-
 export const isSidebarSectionOpen = ( state, section ) =>
-	get( state, `mySites.sidebarSections.${ section }.isOpen` );
+	state?.mySites?.sidebarSections?.[ section ]?.isOpen;
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/my-sites/sidebar/selectors.js]: Fixed.

File: client/state/oauth2-clients/selectors.js [hunk 6]
```diff
@@ -12,5 +6,5 @@ import { get } from 'lodash';
  * @returns {object}          OAuth2 client data
  */
 export const getOAuth2Client = ( state, clientId ) => {
-	return get( state, `oauth2Clients[${ clientId }]`, null );
+	return state?.oauth2Clients?.[ clientId ] ?? null;
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/oauth2-clients/selectors.js]: Fixed.

File: client/state/selectors/get-memberships.js [hunk 7]
```diff
@@ -24,7 +23,7 @@ export default createSelector(
 			return null;
 		}
 
-		const membershipsProducts = get( state, `memberships.productList.items.${ siteId }`, null );
+		const membershipsProducts = state.memberships.productList.items?.[ siteId ] ?? null;
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/selectors/get-memberships.js]: Fixed.

File: client/state/selectors/is-google-my-business-location-connected.js [hunk 8]
```diff
@@ -1,14 +1,11 @@
...
  */
-import { get, find } from 'lodash';
-
 import { getKeyringConnectionsByName } from 'state/sharing/keyring/selectors';
 
 export default function isGoogleMyBusinessLocationConnected( state, siteId ) {
-	const siteKeyrings = get( state, `siteKeyrings.items.${ siteId }`, [] );
-	const googleMyBusinessSiteKeyring = find(
-		siteKeyrings,
+	const siteKeyrings = state?.siteKeyrings?.items?.[ siteId ] ?? [];
```
[the remark this answers is not in the document]
Comment (sgomes) [reply client/state/selectors/is-google-my-business-location-connected.js]: Fixed.

**Comment by sgomes** [comment]
> LGTM 🚢
> 
> It feels like we're too safe in some places (I've offered some suggestions) but as we start modularizing state, it may turn out to be a good idea.

Thank you! Addressed all of them.

As for modularising state, no changes should still be needed in that situation. The idea with modularised state is for state to be ready by the time it's accessed, so a selector should never be in a situation where its part of the state tree is missing. The dependency graph should ensure that state gets initialised as soon as something that needs it gets loaded.

**Comment by tyxla** [comment]
Thanks @sgomes, I think this is still looking good to ship 🚢 , given that the failing tests aren't related 😉

**Comment by sgomes** [comment]
Thanks again, @tyxla! 🙂

**Pull request merged**